app/controllers/steps/check/check_your_answers_controller.rb

```diff
@@ -2,7 +2,11 @@ module Steps
   module Check
     class CheckYourAnswersController < Steps::CheckStepController
       def show
-        @presenter = ResultsPresenter.build(current_disclosure_check)
+        @presenters = current_disclosure_report.check_groups.map do |check_group|
```

Comment:
I suggest creating a separate presenter, named something like CheckAnswersPresenter (or similar) that receives the `current_disclosure_report` and encapsulates all this logic.

That way you still return a single presenter (not a collection of presenters) and is this presenter that knows how to present itself and do the loops etc.

Ideally we would end up with a very similar scenario to what we have now, where:

```html
<dl class="govuk-summary-list">
   <%= render @presenter.summary %>
</dl>
```

Where `@presenter` is the above mentioned `CheckAnswersPresenter`.

Otherwise we risk ending up with too much complexity in the views, and most importantly, not a single spec. The presenter must be tested separately to see if it is doing what is supposed to do.